samples/snippets/create_certificate.py

```diff
@@ -44,30 +40,19 @@ def create_certificate(
         ca_pool_name: set a unique name for the CA pool.
         ca_name: the name of the certificate authority which issues the certificate.
         certificate_name: set a unique name for the certificate.
-        kms_location: Cloud KMS location.
-        key_ring_id: ID of the Cloud KMS key ring.
-        key_id: ID of the key to use.
-        key_version_id: verstion ID of the key to use.
         common_name: a title for your certificate.
         domain_name: fully qualified domain name for your certificate.
         certificate_lifetime: the validity of the certificate in seconds.
+        public_key_bytes: public key used in signing the certificates.
     """
 
-    kmsClient = kms.KeyManagementServiceClient()
     caServiceClient = privateca_v1.CertificateAuthorityServiceClient()
 
-    # To sign and issue a certificate, a public key is essential. Here, we are making use
-    # of Cloud KMS to retrieve an already created public key. For more info, see: https://cloud.google.com/kms/docs/retrieve-public-key.
-    # Generating keys locally is also possible.
+    # The public key used to sign the certificate can be generated using any crypto library/framework.
```

Comment:
Is it reasonable to expect that customers looking at this customer will already have a public key and/or know of a good way to generate one?

If not, or if there are best practices you'd like to show it may be worth pulling the key generation code up into the sample.

Showing that it's possible to use Cloud KMS to create a key also seems valuable to me - maybe you could keep the KMS sample link in the comments?

Reply:
I returned comment about Cloud KMS way.

The reasons why we prefer to generate keys locally:

Keys stored in Cloud KMS are not directly exportable, so the only way a customer can use them is by calling the AsymmetricSign/Verify and Encrypt/Decrypt APIs on those keys 
however, this suffers from two problems:
- most applications that need to use the certificate/key don't support accessing the key like this, but instead require direct access to the key on the filesystem.
- even if they did support that, this would likely introduce too much latency and limit the throughput, so there aren't many scenarios where this would be useful

Leaf certificate keys are almost always generated directly on the machine that will be using them, and never sent anywhere else.